### ISSUE TITLE
Android: startActivity(name, opts) doesn't reflect changing activities on Monitor Session properly

### ISSUE THREAD

**Issue by greybax** [Azure/azure-mobile-engagement-cordova#50]
Call `startActivity` in my app as I described below:
```
$rootScope.$on('$stateChangeSuccess', (e: angular.IAngularEvent, toState, toParams, fromState, fromParams) => {
  Engagement.startActivity(toState.name, extraInfos);
});
```
In debug console `toState.name` passed properly and `Engagement.startActivity(toState.name, {});` finished with `successCallback`. 
But anyway Monitor Session doesn't reflect changes that I've started new activity. In accordance with Monitor Session I'm still stayed in my previous activity. So navigation between the app doesn't reflect any activities while I didn't turn off my mobile and turn on it again. After that on Monitor Session I can see that activity starts properly on the screen which corresponds to activity.

The same thing if I'll do:

```
$rootScope.$on('$stateChangeSuccess', (e: angular.IAngularEvent, toState, toParams, fromState, fromParams) => {
  Engagement.endActivity(() => {
    Engagement.startActivity(toState.name, extraInfos);
  });
});
```

- cordova-android: 6.1.2
- cordova-plugin-ms-azure-mobile-engagement: 3.2.3

**Comment by greybax** [comment]
Figured out that this behavior was caused by issue with too long length of `extraInfos` which I've passed into startActivity as a parameter. I've noticed that it should not be more than 1024 bytes. But anyway I've got `successCallback` from `startActivity` and `endActivity` functions so that behavior a bit confused me.

I'd like to suggest to file another issue about issue that functions `startActivity` and `endActivity` returns successCallback at all time, also if they are wasn't executed successfully.

**Comment by ogoguel** [comment]
Unfortunately, the `startActivity` and `endActivity` in the native SDK only log errors but do not report them, which is why the `failureCallback`is never called.

However, having a length check on the extraInfo parameter before sending it to the SDK could be helpful : will add that in a future release.

BTW in your sample, there's no need to call `endActivity` before starting a new one : it is automatically managed by the native SDKs
